File: openapi/code_samples/java/api_company_search.java
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class CompanyController {

    @GetMapping("/api/company/search")
    public ResponseEntity<?> search(@RequestParam(name = "keyword", defaultValue = "") String keyword) {
        // Your implementation here
        return ResponseEntity.ok().build();
    }
}

File: openapi/code_samples/csharp/api_company_search.cs
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;

[ApiController]
[Route("api/[controller]")]
public class CompanyController : ControllerBase
{
    [HttpGet]
    [Route("search")]
    [SwaggerOperation(Summary = "Retrieves the top 10 companies based on CUI or name as the keyword.")]
    [ProducesResponseType(typeof(SearchCompanyResponse[]), 200)]
    [ProducesResponseType(typeof(Fault), 400)]
    public IActionResult Search([FromQuery(Name = "keyword")] string keyword = "")
    {
        // Your implementation here
        return Ok();
    }
}
